agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/LiveMetricsSpanProcessor.java

```diff
@@ -21,95 +21,41 @@
-import javax.annotation.Nullable;
 
 public class LiveMetricsSpanProcessor implements SpanProcessor {
 
-  private static final ClientLogger LOGGER = new ClientLogger(AzureMonitorTraceExporter.class);
-
   private final QuickPulse quickPulse;
-  private final String instrumentationKey;
+  private final SpanDataMapper mapper;
 
-  // todo this is never used.
-  public LiveMetricsSpanProcessor(
-      HttpPipeline httpPipeline,
-      URL endpointUrl,
-      String instrumentationKey,
-      @Nullable String roleName,
-      @Nullable String roleInstance,
-      String sdkVersion) {
-    this.instrumentationKey = instrumentationKey;
-    quickPulse =
-        QuickPulse.create(
-            httpPipeline,
-            () -> endpointUrl,
-            () -> instrumentationKey,
-            roleName,
-            roleInstance,
-            true,
-            sdkVersion);
+  public LiveMetricsSpanProcessor(QuickPulse quickPulse, SpanDataMapper mapper) {
+    this.quickPulse = quickPulse;
+    this.mapper = mapper;
   }
 
   @Override
-  public void onStart(Context context, ReadWriteSpan readWriteSpan) {
-    SpanData spanData = readWriteSpan.toSpanData();
-    SpanKind kind = spanData.getKind();
-    String instrumentationName = spanData.getInstrumentationLibraryInfo().getName();
-
-    List<TelemetryItem> telemetryItems = new ArrayList<>();
-    if (kind == SpanKind.INTERNAL) {
-      if (instrumentationName.startsWith("io.opentelemetry.spring-scheduling-")
-          && !spanData.getParentSpanContext().isValid()) {
-        // if (!span.getParentSpanContext().isValid()) {
-        // TODO (trask) AI mapping: need semantic convention for determining whether to map INTERNAL
-        // to request or dependency (or need clarification to use SERVER for this)
-        TelemetryMappingHelper.addRequestTelemetryItem(
-            spanData, instrumentationKey, telemetryItems);
-      } else {
-        TelemetryMappingHelper.addRemoteDependencyTelemetryItem(
-            spanData, true, instrumentationKey, telemetryItems);
-      }
-    } else if (kind == SpanKind.CLIENT || kind == SpanKind.PRODUCER) {
-      TelemetryMappingHelper.addRemoteDependencyTelemetryItem(
-          spanData, false, instrumentationKey, telemetryItems);
-    } else if (kind == SpanKind.CONSUMER
-        && "receive".equals(spanData.getAttributes().get(SemanticAttributes.MESSAGING_OPERATION))) {
-      TelemetryMappingHelper.addRemoteDependencyTelemetryItem(
-          spanData, false, instrumentationKey, telemetryItems);
-    } else if (kind == SpanKind.SERVER || kind == SpanKind.CONSUMER) {
-      TelemetryMappingHelper.addRequestTelemetryItem(spanData, instrumentationKey, telemetryItems);
-    } else {
-      throw LOGGER.logExceptionAsError(new UnsupportedOperationException(kind.name()));
-    }
-
-    for (TelemetryItem telemetryItem : telemetryItems) {
-      quickPulse.add(telemetryItem);
-    }
-  }
+  public void onStart(Context context, ReadWriteSpan readWriteSpan) {}
```

Comment:
onStart is empty?

Reply:
ya, live metrics are reported on span end